server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -478,6 +502,77 @@ private RestResponse getDetailResponse(Cache<?, ?> cache) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String cacheName = request.variables().get("cacheName");
+
+      responseBuilder.contentType(APPLICATION_JSON);
+      if (!invocationHelper.getRestCacheManager().getInstance().getCacheConfigurationNames().contains(cacheName)) {
+         responseBuilder.status(NOT_FOUND).build();
+      }
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
+      if (cache == null)
+         return notFoundResponseFuture();
+
+      Configuration cacheConfiguration = SecurityActions.getCacheConfiguration(cache.getAdvancedCache());
+      List<String> attributes = new ArrayList<>();
+      mutableAttributes(cacheConfiguration, attributes, null);
+      return asJsonResponseFuture(Json.make(attributes));
+   }
+
+   private static void mutableAttributes(ConfigurationElement<?> element, List<String> attributes, String prefix) {
+      prefix = prefix == null ? "" : element.name();
+      for (Attribute<?> attribute : element.attributes().attributes()) {
+         if (!attribute.isImmutable()) {
+            attributes.add(prefix + "." + attribute.getAttributeDefinition().name());
+         }
+      }
+      for (ConfigurationElement<?> child : element.children()) {
+         mutableAttributes(child, attributes, prefix);
+      }
+   }
+
+   private CompletionStage<RestResponse> getCacheConfigMutableAttribute(RestRequest request) {
+      String attributeName = request.getParameter("attribute-name");
+      String cacheName = request.variables().get("cacheName");
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
+      if (cache == null) {
+         return notFoundResponseFuture();
+      }
+      Configuration cacheConfiguration = SecurityActions.getCacheConfiguration(cache.getAdvancedCache());
+      Attribute<?> attribute = cacheConfiguration.findAttribute(attributeName);
+      if (attribute.isImmutable()) {
+         return responseFuture(BAD_REQUEST);
+      } else {
+         return asJsonResponseFuture(Json.make(String.valueOf(attribute.get())));
+      }
+   }
+
+   private CompletionStage<RestResponse> setCacheConfigMutableAttribute(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String attributeName = request.getParameter("attribute-name");
+      String attributeValue = request.getParameter("attribute-value");
+      String cacheName = request.variables().get("cacheName");
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
+      if (cache == null) {
+         return notFoundResponseFuture();
+      }
+      Configuration configuration = new ConfigurationBuilder().read(SecurityActions.getCacheConfiguration(cache.getAdvancedCache())).build();
+      Attribute<?> attribute = configuration.findAttribute(attributeName);
```

Comment:
We should probably return a 400 here if `if (attribute.isImmutable()) {` in order to avoid an update that we know will fail.